Creates a zipfile of all items

AndrewGorton opened the pull request:
This creates a zipfile on the fly containing all the register entries.

**For larger registers, this may not be the most efficient method.**

See http://openregister.github.io/specification/#download for a bit more details.

~~Depends upon the https://github.com/openregister/indexer/pull/35 for the timestamps in the Entry files, and probably makes sense to merge after https://github.com/openregister/presentation/pull/163.~~

Commit:
Initial attempt which creates zip of all items

Commit:
Added the Certificate Transparency proof to the download archive.

Commit:
Added check to ensure the proofs file is in the archive

philandstuff commented on src/main/java/uk/gov/register/presentation/Proofs.java:
```diff
@@ -0,0 +1,15 @@
+        return certificateTransparencySignedTreeHead;
+    }
+
+    public void setCertificateTransparencySignedTreeHead(SignedTreeHead certificateTransparencySignedTreeHead) {
```
our jackson data objects so far have used constructor injection rather than setter injection. could this class do the same for consistency?

Commit:
Changed to constructor injection for the data object

philandstuff commented on src/main/java/uk/gov/register/presentation/resource/DataResource.java:
```diff
@@ -1,41 +1,96 @@
     }
 
     @GET
     @Path("/download")
```
comment to capture that we still need to confirm URL structure

philandstuff commented:
:+1: looks good so far but not merging per request

AndrewGorton commented on build.gradle:
```diff
@@ -87,6 +87,9 @@ test {
         exceptionFormat = 'full'
     }
+    afterTest { desc, result ->
+        println "Executing test ${desc.name} [${desc.className}] with result: ${result.resultType}"
```
For posterity: This is here so that I can check my test(s) actually ran. It emits the test name and result into the build log. Otherwise the only time I am sure my test ran is by deliberately making it fail.

I know that I could run gradle in INFO and DEBUG mode which would also display the test output, but it also fills the console with additional buildtime fluff.

Commit:
Split the archive creation out from the resource. Reinstated the download help, and added
a link to point to the URL for downloading the entire register as a zip archive.

Commit:
Changed item filenames to be the item hash.
Prevent duplicate items (previously, underlying library throws an exception).
Started adding entries to include item hash and entry number.

Commit:
Merge branch 'master' of github.com:openregister/presentation into get-raw-entries

Commit:
Added the register details to the download archive.
Changed the filename to include the timestamp of the download.

Commit:
Now reads the leaf_input from the database.

Commit:
Tests fixed to include leaf_input.

Commit:
Merge branch 'master' of github.com:openregister/presentation into get-raw-entries
